install.sh

```diff
@@ -1,81 +1,166 @@
 #!/bin/bash
-GOODCOLOR='\033[1;36m'
-WARNCOLOR='\033[1;31m'
+
 NOCOLOR='\033[0m'
-GOODPREFIX="${GOODCOLOR}INSTALLER:"
-WARNPREFIX="${WARNCOLOR}ERROR:"
+LOGCOLOR='\033[1;36m'
+PASSCOLOR='\033[1;32m'
+WARNCOLOR='\033[1;31m'
 
-# Sane installation defaults for no argument cases
-CATKIN_DIR=~/sub_ws
-REQUIRED_OS="trusty"
+LOGPREFIX="${LOGCOLOR}INSTALLER:"
+WARNPREFIX="${WARNCOLOR}ERROR:"
+PASSTEXT="${PASSCOLOR}PASS"
+FAILTEXT="${WARNCOLOR}FAIL"
 
 instlog() {
-    printf "$GOODPREFIX $@ $NOCOLOR\n"
+    printf "$LOGPREFIX $@ $NOCOLOR\n"
 }
 
 instwarn() {
     printf "$WARNPREFIX $@ $NOCOLOR\n"
 }
 
+
+instpass() {
+    printf "$PASSTEXT $NOCOLOR"
+}
+
+
+instfail() {
+    printf "$FAILTEXT $NOCOLOR"
+}
+
+check_host() {
+
+    # Attempts to ping a host to make sure it is reachable
+    HOST="$1"
+
+    HOST_PING=$(ping -c 2 $HOST 2>&1 | grep "% packet" | cut -d" " -f 6 | tr -d "%")
+    if ! [ -z "${HOST_PING}" ]; then
+
+        # Uses packet loss percentage to determine if the connection is strong
+        if [ $HOST_PING -lt 25 ]; then
+
+            # Will return true if ping was successful and packet loss was below 25%
+            return `true`
+        fi
+    fi
+    return `false`
+}
+
```

Comment:
I saw no real need to add more complexity. "git clone bleh" done. Although, it may be worth to add it back for the recursive option.